azurerm/resource_arm_log_profile_test.go

```diff
@@ -0,0 +1,318 @@
+	rs := acctest.RandString(10)
+	config := testAccAzureRMLogProfile_basic(ri, rs, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMLogProfileDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLogProfileExists(resourceName),
+					testCheckAzureRMLogProfileDisappears(resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureRMLogProfileDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ArmClient).logProfilesClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_log_profile" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resp, err := client.Get(ctx, name)
+		if err != nil {
+			return nil
+		}
+
+		return fmt.Errorf("Log Profile still exists:\n%#v", *resp.ID)
+	}
+
+	return nil
+}
+
+func testCheckAzureRMLogProfileExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		client := testAccProvider.Meta().(*ArmClient).logProfilesClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		name := rs.Primary.Attributes["name"]
+		resp, err := client.Get(ctx, name)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Log Profile %q does not exist", name)
+			}
+
+			return fmt.Errorf("Bad: Get on logProfilesClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMLogProfileDisappears(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		name := rs.Primary.Attributes["name"]
+
+		client := testAccProvider.Meta().(*ArmClient).logProfilesClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		_, err := client.Delete(ctx, name)
+		if err != nil {
+			return fmt.Errorf("Error deleting Log Profile %q: %+v", name, err)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMLogProfile_basic(rInt int, rString string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+	name     = "acctest%d-rg"
+	location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+	name                     = "%s"
+	resource_group_name      = "${azurerm_resource_group.test.name}"
+	location                 = "${azurerm_resource_group.test.location}"
+	account_tier             = "Standard"
+	account_replication_type = "GRS"
+}
+	
+resource "azurerm_log_profile" "test" {
+	name = "basic"
```

Comment:
Append random number to `name` to prevent name conflict when running tests in parallel.

Reply:
The acceptance tests are combined into one run now. Azure doesn't allow for more than one log profile in a subscription so these tests shouldn't run in parallel.